maxi-blocks/maxi-blocks
# Column Maxi - Fit content

Adds 'Fit content' option on Column Maxi.

This options helps to fix this situation when trying to dispose some elements in a row with same gap is hard. Top container is with the current column, second one is with the `Fit content` new option (on Column, Column Settings accordion) that keeps same distance between elements:

https://user-images.githubusercontent.com/50477222/156534267-53912afd-98ae-4fc3-9cfd-81397835edbd.mp4

Important ❗ 

In order to do this implementation, I had to cheat a bit. The 'columnSize' attributes, which were a number type, need to become string. For it, I had to create a `deprecation` on `src/blocks/column-maxi/column-maxi.js` which migrates the type of 'columnSize'. Onces all this migration will be done from the Library that has been already done, that part of code will disappear. 

To test:
1. Creating a Grid the columns `Fit content` action works good.
2. If pasting a content [like this one](https://github.com/yeahcan/maxi-blocks/files/8176543/editor.txt) where includes a Grid with Column that uses 'columnSize' attributes with numbers (not strings) it migrates correctly saving the 'columnSize' attributes as strings. Reload the page to check it works correctly, and use this command `wp.data.select('core/block-editor').getBlockAttributes(wp.data.select('core/block-editor').getSelectedBlockClientId())` in console to request for block attributes selecting the columns of the pattern and then check the attributes for 'columnSize' (`column-size-${breakpoint}`) are strings or undefined.
